packages/rehype-parse/types/index.d.ts

```diff
@@ -0,0 +1,72 @@
+// TypeScript Version: 3.5
+import {Parser, Plugin} from 'unified'
+import {Node, Parent} from 'unist'
+
+declare namespace rehypeParse {
+  interface Parse extends Plugin<[RehypeParserOptions?]> {
+    Parser: Parser
+  }
+
+  interface RehypeParserOptions {
+    /**
+     * Specify whether to parse a fragment, instead of a complete document.
+     * In document mode, unopened `html`, `head`, and `body` elements are opened in
+     * just the right places.
+     *
+     * @default false
+     */
+    fragment?: boolean
+
+    /**
+     * > ⚠️ rehype is not an XML parser.
+     * > It support SVG as embedded in HTML, but not the features available in the rest
+     * > of XML/SVG.
+     * > Passing SVG files could strip useful information, but fragments of modern SVG
+     * > should be fine.
+     *
+     * Which space the document is in.
+     *
+     * If an `svg` element is found in the HTML space, `parse` automatically
+     * switches to the SVG space when [**entering**](https://github.com/syntax-tree/unist#enter)
+     * the element, and switches
+     * back when [**exiting**](https://github.com/syntax-tree/unist#exit).
+     *
+     * **Note**: make sure to set `fragment: true` if `space: 'svg'`.
+     *
+     * @default 'html'
+     */
+    space?: 'html' | 'svg'
```

Comment:
this inherited from https://github.com/syntax-tree/hast-util-from-parse5#optionsspace
it may be good to have typings for `hast-util-from-parse5` to leverage here.

Reply:
Yes, it’s a good idea, but at the same time it’s not. I created syntax-tree/hast-util-from-parse5#10, because I do agree with you. However, as can be seen in that PR description, this requires to pull in a bunch of dependencies users need to install manually to consume these types.

The deeper issue is unifiedjs/unified#45

A proper solution would be to move type dependencies into `dependencies`, not `devDependencies`.

A workaround would be to keep this PR as-is, and not bother users with these quirks coming from `hast-util-from-parse5`.

Reply:
syntax-tree/hast-util-from-parse5#10 as been merged and unifiedjs/unified#45 resolved.
Thanks for the help on those!

would the type be able to be reused now? or is there another blocker I missed?

Reply:
I think I've got everything I need to resolve the remaining issues. I'll probably fix it this weekend.